packaging_automation/update_docker.py

```diff
@@ -0,0 +1,147 @@
+
+    if postgres_version:
+        update_pkgvars(project_version, template_path, pkgvars_file, postgres_version)
+
+    persisted_postgres_version = read_postgres_version(pkgvars_file)
+
+    update_docker_file_for_latest_postgres(project_version, template_path, exec_path, persisted_postgres_version)
+    update_regular_docker_compose_file(project_version, template_path, exec_path)
+    update_docker_file_alpine(project_version, template_path, exec_path, persisted_postgres_version)
+    update_docker_file_for_postgres12(project_version, template_path, exec_path)
+    update_changelog(project_version, exec_path, postgres_version)
+
+
+def read_postgres_version(pkgvars_file: str) -> str:
+    if os.path.exists(pkgvars_file):
+        with open(pkgvars_file, "r") as reader:
+            content = reader.read()
+            lines = content.splitlines()
+            for line in lines:
+                if line.startswith("latest_postgres_version"):
+                    line_parts = line.split("=")
+                    if len(line_parts) != 2:
+                        raise ValueError("keys and values should be seperated with '=' sign")
+                    else:
+                        postgres_version = line_parts[1]
+                else:
+                    raise ValueError("pkgvars file should include a line with key latest_postgres_version")
+    else:
+        # Setting it because pkgvars does not exist initially
+        postgres_version = "13.2"
+    return postgres_version
+
+
+def update_pkgvars(project_version: str, template_path: str, pkgvars_file: str, postgres_version: str):
+    if postgres_version:
+        content = process_template_file(project_version, template_path, "docker-pkgvars.tmpl", postgres_version)
+        with open(pkgvars_file, "w") as writer:
+            writer.write(content)
+
+
+CHECKOUT_DIR = "docker_temp"
+
+
+def create_pr():
+    g = Github(github_token)
+    repository = g.get_repo(f"{REPO_OWNER}/{PROJECT_NAME}")
+    repository.create_pull(title=f"Bump Citus to {args.prj_ver}", base="master",
+                           head=pr_branch, body="")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--prj_ver', required=True)
+    parser.add_argument('--gh_token', required=True)
+    parser.add_argument('--postgres_version')
+    parser.add_argument('--is_test', action="store_true")
+    args = parser.parse_args()
+
+    execution_path = f"{os.getcwd()}/{CHECKOUT_DIR}"
+    github_token = args.gh_token
+
+    tools_path = os.getcwd()
+
+    initialize_env(execution_path, PROJECT_NAME, CHECKOUT_DIR)
+    os.chdir(execution_path)
+    run("git checkout master")
+    pr_branch = f"release-{args.prj_ver}-{uuid.uuid4()}"
+    run(f"git checkout -b {pr_branch}")
+    update_all_docker_files(args.prj_ver, tools_path, execution_path, args.postgres_version)
+    run("git add .")
+    run(f'git commit -m "Bump to version {args.prj_ver}"')
```

Comment:
```suggestion
    run(f'git commit -a -m "Bump to version {args.prj_ver}"')
```

Reply:
Also, we can put commit message into a variable and pass it to `create_pgxn_pr` to use it as pr title too.

Reply:
I parameterized commit message and used it as pr title for both update docker and update pgxn

Reply:
I also meant removing `git add .` & using `-a` in `git commit` command

Reply:
git commit -a does not add new files so I need to add git add . which covers new files as well

Reply:
https://git-scm.com/docs/git-commit
-a
--all
Tell the command to automatically stage files that have been modified and deleted, but **new files you have not told Git about are not affected.**

Reply:
Will we add pkgvars to docker repo at some point ?
If yes, then not resolving this comment so that we don't forget to fix this after introducing pkgvars file to docker repo.

Reply:
I think we can keep "git add ."   even after merging pkgvars file. It has no side effects I think

Reply:
yeah you're right, just to be consistent with the other places doing `git commit -a` instead of `git add .` & `git commit`

Reply:
ok I will evaluate in the next PR and use git add . & git commit pair